src/components/SingleThought.jsx

```diff
@@ -0,0 +1,67 @@
+import { useEffect, useState } from "react"
+import moment from "moment"
+import "../styling/thought.css"
+
+export const SingleThought = ({ eachThought, onLikeChange }) => {
+	const [like, setLike] = useState(false)
+	const [numberLikes, setNumberLikes] = useState(eachThought.hearts)
+
+	const LIKE_API = `https://happy-thoughts-ux7hkzgmwa-uc.a.run.app/thoughts/${eachThought._id}/like`
+
+	useEffect(
+		() => {
+			const likedThought =
+				JSON.parse(localStorage.getItem("likedThought")) || []
+			if (likedThought.includes(eachThought._id)) {
+				setLike(true)
+			}
+		},
+		[eachThought._id],
+		like
+	)
```

Comment:
since you are using localStorage , you could disable the like button if the user try to put a like on a post he already liked. (if id of new like post match with an id already in the like array then don't take it)